app/src/main/java/com/getsensibill/spendmanager/demo/subclass/SubclassKotlinActivity.kt

```diff
@@ -0,0 +1,119 @@
+package com.getsensibill.spendmanager.demo.subclass
+
+import android.os.Bundle
+import com.getsensibill.spendmanager.demo.databinding.ActivitySubclassKotlinBinding
+import com.getsensibill.web.data.UiFinishReason
+import com.getsensibill.web.data.configuration.NavigationIntent
+import com.getsensibill.web.data.configuration.ProgrammaticTheme
+import com.getsensibill.web.data.models.Brand
+import com.getsensibill.web.ui.WebUiActivity
+import com.getsensibill.web.ui.WebUiFragment
+import com.getsensibill.web.ui.WebUiNetworkErrorFragment
+import timber.log.Timber
+
+/**
+ * Subclassing the WebUiActivity
+ * <p>
+ * When you subclass the WebUiActivity, you do not need to override anything for it to work.
+ * It will use all the defaults provided by the [WebUiActivity] and the [WebUiFragment].
+ *
+ * If you are using your own view, then its important to override the [callSetContentView] and
+ * the [webContainerId] so that the [WebUiActivity] knows where to place the [WebUiFragment].
+ */
+class SubclassKotlinActivity : WebUiActivity(), WebUiFragment.Listener {
+    private lateinit var binding: ActivitySubclassKotlinBinding
+
+    /**
+     * Not required override.
+     *
+     * Can be used to override the setContentView, so that it uses your provided
+     * activities view, instead of the default [WebUiActivity] view.
+     */
+    override fun callSetContentView() {
+        binding = ActivitySubclassKotlinBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+    }
+
+    /**
+     * Not required override, unless you are overriding [callSetContentView]
+     *
+     * If we are overriding the setContentView, we also need to set the container id
+     * which will contain the [WebUiFragment].
+     */
+    override val webContainerId: Int
+        get() = binding.webContainer.id
+
+    /**
+     * Not required override.
+     *
+     * If you want to change the default navigation when first loading into Spend Manager,
+     * you can override this and provide a different [NavigationIntent]. Alternatively,
+     * you can also apply this as an argument when overriding [createWebUiFragment].
+     */
+    override val webInitialNavigation: NavigationIntent
+        get() = NavigationIntent.DASHBOARD
+
+    /**
+     * Not required override.
+     *
+     * Used to create the [WebUiFragment] held in the Activity. Common reason for overriding
+     * this would be if there are arguments that are needed to be supplied to the fragment.
+     */
+    override fun createWebUiFragment(): WebUiFragment = WebUiFragment().apply {
```

Comment:
What do you think our suggested implementation here should be?
We have two options:
1) What you've done here, create the web fragment then pass arguments
2) What we've done in the `WebUiBetaWrapActivity` - call `super.create...` and then `.apply` on that to add arguments

Currently I like what we do in the beta subclass:
```kotlin
override fun createWebUiFragment(): WebUiFragment = super.createWebUiFragment().apply {
    arguments = (arguments ?: Bundle()).apply {
        // add arg
    }
}
```
because that will allow the client to inherit whatever customizations we have and add on top, rather than resetting / creating a brand new fragment.

In terms of actual effects here, I think if they create their own rather than using super, overriding `webInitialNavigation` and `programmaticTheme` won't work because they're given to the fragment in the super.

This discussion also applies to the create error fragment method

Reply:
I agree, that makes sense. Updated to use super.